Add AStyle definition file and travis astyle check

0xc0170 opened the pull request:
@geky Your latest PR was going the good direction. I defined astyle rc based on the style we have in this codebase and added it to Travis

This is work in progress, as we need to actually define what we should be ignoring (I defined few I could think of, however there are external code in our codebase like SDK, network stacks, etc). Therefore this needs more work to be done (I could not find actually that --exclude would support wildcards?? for astyle).

I run locally on drivers only, lot of changes done in files. Thus travis should fail and point to all files that need changes.

geky commented on .travis.yml:
```diff
@@ -93,6 +93,24 @@ matrix:
 
+    - env:
+        - TEST="astyle"
+      before_install:
```
Unfortunately, Travis doesn't inherit from multiple before_installs.

If you want the status update + pipefail + caching (and probably just future compatibility), you'll need to copy them from the global before_install or move this to the install. Maybe a YAML reference will work?

0xc0170 replied:
will check, and fix somehow.. I noticed that using default version it complains about options not available but their webpage lists them, so need to get update somehow , will fix

geky commented on .travis.yml:
```diff
@@ -93,6 +93,24 @@ matrix:
+      install:
+        - sudo apt-get install astyle
+      script:
+        - export ARTISTIC_STYLE_OPTIONS=".astylerc" && astyle `find . | egrep "\.c|\.h|\.hpp|\.cpp"` > astyle.out
```
tiny nit for if you change the file again: find can do this for you:
``` diff
find . | egrep "\.c|\.h|\.hpp|\.cpp"
find -regex '.*\.\(h\|c\|hpp\|cpp\)'
```

0xc0170 commented on .travis.yml:
```diff
@@ -93,6 +95,20 @@ matrix:
+      install:
+        - sudo apt-get install astyle
+      script:
+        - export ARTISTIC_STYLE_OPTIONS=".astylerc" && $ find -regex '.*\.\(h\|c\|hpp\|cpp\)' -type f ! -path "./cmsis/*" ! -path "./features/mbedtls/*"  ! -path "./features/FEATURE_LWIP/lwip/*"  ! -path "./rtos/TARGET_CORTEX/rtx5/RTX/*"  ! -path "./BUILD/*"  ! -path "./tools/*" | xargs -n 100 -I {} bash -c "astyle --options=.astylerc \"{}\"" > astyle.out
```
@geky what magic can we use to exclude some directories ? I defined only some generic ones, but lot of targets contain their own drivers.

I tried to use --exclude, but always end up with error unmatched pattern from astyle 😢 

Shall we define exclude paths in a file, or provide .astyleignore file in those folders?

0xc0170 commented:
Note: I've noticed AStyle in ubuntu that is used in travis is around v2.05, the reference manual on their official webpage is for 3.x :-) their latest version is published for unstable versions at the moment 😀

0xc0170 commented:
Looking at travis, the command seems to be working, I can see lot of diffs there for files that needs to be changed.

Based on the output, here are proposed exclude dirs (along the ones that I already defined):

```

.\features\FEATURE_BLE\targets\TARGET_CORDIO\stack\cordio_stack
.\features\FEATURE_BLE\targets\TARGET_Maxim\exactLE
.\features\FEATURE_BLE\targets\TARGET_NORDIC\TARGET_MCU_NRF51822\sdk\
.\features\FEATURE_LWIP\lwip-interface\lwip

.\features\unsupported\

.\targets\TARGET_Analog_Devices\TARGET_ADUCM302X\TARGET_ADUCM3029\bsp
.\targets\TARGET_Analog_Devices\TARGET_ADUCM4X50\TARGET_ADUCM4050\bsp
.\targets\TARGET_ARM_SSG\TARGET_BEETLE\cordio
.\targets\TARGET_Atmel\TARGET_SAM_CortexM0P\config
.\targets\TARGET_Atmel\TARGET_SAM_CortexM0P\drivers
.\targets\TARGET_Atmel\TARGET_SAM_CortexM0P\utils
.\targets\TARGET_Atmel\TARGET_SAM_CortexM4\config\
.\targets\TARGET_Atmel\TARGET_SAM_CortexM4\drivers\
.\targets\TARGET_Atmel\TARGET_SAM_CortexM4\services\
.\targets\TARGET_Atmel\TARGET_SAM_CortexM4\utils\cmsis\TARGET_SAMG55\include
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_K66F\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_K82F\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_KL27Z\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_KL43Z\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_KL82Z\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_KW24D\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_KW41Z\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_MCU_K22F\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_MCU_K24F\drivers
.\targets\TARGET_Freescale\TARGET_MCUXpresso_MCUS\TARGET_MCU_K64F\drivers
.\targets\TARGET_Maxim\TARGET_MAX32600\device
.\targets\TARGET_Maxim\TARGET_MAX32610\device
.\targets\TARGET_Maxim\TARGET_MAX32620\device
.\targets\TARGET_Maxim\TARGET_MAX32625\device
.\targets\TARGET_Maxim\TARGET_MAX32625\mxc
.\targets\TARGET_Maxim\TARGET_MAX32630\device
.\targets\TARGET_Maxim\TARGET_MAX32630\mxc
.\targets\TARGET_NORDIC\TARGET_MCU_NRF51822\Lib\nordic_sdk
.\targets\TARGET_NORDIC\TARGET_NRF5\TARGET_MCU_NRF51822_UNIFIED\sdk
.\targets\TARGET_NORDIC\TARGET_NRF5\TARGET_MCU_NRF52832\sdk
.\targets\TARGET_NORDIC\TARGET_NRF5\TARGET_MCU_NRF52840\sdk
.\targets\TARGET_NORDIC\TARGET_NRF5\TARGET_SDK11
.\targets\TARGET_NORDIC\TARGET_NRF5\TARGET_SDK13
.\targets\TARGET_NUVOTON\TARGET_M451\device\StdDriver
.\targets\TARGET_NUVOTON\TARGET_M480\device\StdDriver
.\targets\TARGET_NUVOTON\TARGET_NANO100\device\StdDriver
.\targets\TARGET_NUVOTON\TARGET_NUC472\device\StdDriver
.\targets\TARGET_NXP\TARGET_MCUXpresso_MCUS\TARGET_LPC54114\drivers
.\targets\TARGET_NXP\TARGET_MCUXpresso_MCUS\TARGET_LPC546XX\drivers
.\targets\TARGET_Realtek\TARGET_AMEBA\sdk
.\targets\TARGET_RENESAS\TARGET_RZ_A1XX\TARGET_GR_LYCHEE\device\inc
.\targets\TARGET_RENESAS\TARGET_RZ_A1XX\TARGET_RZ_A1H\device\inc
.\targets\TARGET_RENESAS\TARGET_RZ_A1XX\TARGET_VK_RZ_A1H\device\inc
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\emlib
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32GG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32HG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32LG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32PG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32PG12\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32WG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFM32ZG\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFR32MG1\device
.\targets\TARGET_Silicon_Labs\TARGET_EFM32\TARGET_EFR32MG12\device
.\targets\TARGET_Silicon_Labs\TARGET_SL_RAIL\efr32-rf-driver
.\targets\TARGET_STM\TARGET_STM32F0\device
.\targets\TARGET_STM\TARGET_STM32F1\device
.\targets\TARGET_STM\TARGET_STM32F2\device
.\targets\TARGET_STM\TARGET_STM32F3\device
.\targets\TARGET_STM\TARGET_STM32F4\device
.\targets\TARGET_STM\TARGET_STM32F7\device
.\targets\TARGET_STM\TARGET_STM32L0\device
.\targets\TARGET_STM\TARGET_STM32L1\device
.\targets\TARGET_STM\TARGET_STM32L4\device
.\targets\TARGET_TOSHIBA\TARGET_TMPM066\Periph_Driver
.\targets\TARGET_WIZNET\TARGET_W7500x\W7500x_Peripheral_Library
```

geky commented on .travis.yml:
```diff
@@ -93,6 +95,20 @@ matrix:
         - bash -c "$STATUS" success "Local $NAME testing has passed"
 
+    - env:
+        - TEST="astyle"
```
Did you mean "NAME" here?

0xc0170 replied:
fixed

geky commented on .travis.yml:
```diff
@@ -93,6 +95,20 @@ matrix:
+      install:
+        - sudo apt-get install astyle
+      script:
+        - export ARTISTIC_STYLE_OPTIONS=".astylerc" && find -regex '.*\.\(h\|c\|hpp\|cpp\)' -type f ! -path "./cmsis/*" ! -path "./features/mbedtls/*"  ! -path "./features/FEATURE_LWIP/lwip/*"  ! -path "./rtos/TARGET_CORTEX/rtx5/RTX/*"  ! -path "./BUILD/*"  ! -path "./tools/*" | xargs -n 100 -I {} bash -c "astyle --options=.astylerc \"{}\"" > astyle.out
```
`ARTISTIC_STYLE_OPTIONS=".astylerc"` could go in the env

geky commented on .astyleignore:
```diff
@@ -0,0 +1,76 @@
+features\FEATURE_BLE\targets\TARGET_Maxim\exactLE
+features\FEATURE_BLE\targets\TARGET_NORDIC\TARGET_MCU_NRF51822\sdk\
+features\FEATURE_LWIP\lwip-interface\lwip
+features\unsupported\
```
Are your slashes backwards?

0xc0170 replied:
Fixed

0xc0170 commented on .astylerc:
```diff
@@ -0,0 +1,39 @@
+attach-classes
+
+# Folders to be excluded
+# exclude=cmsis
```
will remove

0xc0170 commented:
I sent docs update to cover this addition and automatic check - https://github.com/ARMmbed/Handbook/pull/400

Once travis finishes, we review all changes and I can add a new commits here fixing the code until we make AStyle green

0xc0170 commented:
The last travis run works with astyle, we can see all the styling issues now displayed  t here. I'll add commits to fix those issues to make travis green

geky commented on TESTS/events/queue/main.cpp:
```diff
@@ -31,54 +31,60 @@ using namespace utest::v1;
 
-void func1(int a0) {
+void func1(int a0)
+{
```
I was curious about this so I tried astyle with both brace styles:
```
with style=kr:     1841070 line changes
with style=attach: 1819522 line changes
```

It would actually take fewer code changes to change the existing KR braces than to standardize on KR. We should reconsider our stance on brace placement.

I've never understood why our brace style is inconsistent between functions and every other syntactic construct.

0xc0170 replied:
> It would actually take fewer code changes to change the existing KR braces than to standardize on KR. We should reconsider our stance on brace placement.

I have testing online compiler - we need to match what we have here and what online compiler does. It seems it defines a rule to actually attach `{` in the header files, will run few more tests and get its options that it defines

To have to have both aligned.

0xc0170 replied:
I updated this PR, attached C++ as the online compiler does.

pan- replied:
This contradict the guidelines and guidelines are made to be followed; the purpose of this PR is to enforce them. If the online compiler is wrong; fix the online compiler. Not the other way around.

geky replied:
Updated, KR style braces are still the minority, although only slightly:
```
with style=kr:     1819759  line changes
with style=attach: 1819522 line changes
```

I'm suggesting we change the guidelines to match the style of code users are contributing. I had no idea C++ inlines are a separate style, and if we don't know what our rules are on braces how are our users supposed to?

I mean what is this:

``` cpp
namespace hi {

class Thing {
    int dothing() {
        if (thing) {
            blahblahblah;
        } else {
            blahblahblah;
        }
    }
};

int dothing()
{
    while (thing) {
        blahblahblah;
    }
}

}
```
What's the reason for not making squiggly brackets consistent?

kjbracey-arm commented:
Nanostack and the related "COMMON_PAL" code are subtrees from master external repos - I think it would be less disruptive to format them over there then merge, initially. Incoming subtree merges can certainly be checked for formatting.

0xc0170 commented:
> Nanostack and the related "COMMON_PAL" code are subtrees from master external repos - I think it would be less disruptive to format them over there then merge, initially. Incoming subtree merges can certainly be checked for formatting.

I'll remove nanostack and common pal here

kjbracey-arm commented:
To be a little more precise - these are from out-of-tree:

* FEATURE_NANOSTACK/coap-service
* FEATURE_NANOSTACK/sal-stack-nanostack
* FEATURE_COMMON_PAL/<almost everything>

These are mbed OS specific components, mastered in-tree and can be reformatted:

* FEATURE_COMMON_PAL/nanostack-hal-mbed-cmsis-rtos
* FEATURE_NANOSTACK/mbed-mesh-api
* FEATURE_NANOSTACK/nanostack-interface
* FEATURE_NANOSTACK/targets

0xc0170 commented:
Comparing to the online IDE formatter, what we do in the codebase and define even in our code style:

```
attach-namespaces
attach-classes
```
I rebased this, to clean the history a bit and reapply the coding style.

0xc0170 commented:
Rebased, I noticed v3.0 had some bugs and was formatting code that was valid, locally using v3.1 that does not show these issues (related to code like (int)-1 where it ends up with (int) - 1). Travis should fetch v3.1 now.

pan- commented on features/FEATURE_BLE/ble/ArrayView.h:
```diff
@@ -83,17 +83,16 @@ struct ArrayView {
     template<size_t Size>
-    ArrayView(T (&elements)[Size]):
-		_array(elements), _size(Size) { }
+    ArrayView(T(&elements)[Size]):
```
Where is it mentioned that we should remove space in reference to array declaration ?

0xc0170 replied:
I'll reply here, there are few similar comments. From the documentation:

> Artistic Style cannot always determine the usage of symbols with more than one meaning. For example an asterisk (*) can be multiplication, a pointer, or a pointer dereference. The "&" and "&&" symbols are a similar problem.
> If a symbol is being padded incorrectly, padding it manually may fix the problem. If it is still being padded incorrectly, then disabling the formatting may be necessary. To avoid having to use the "disable block" tags above, a single line disable is available.

😞 

I'll play with options around padding. I fear we leave padding out of it (3 pad rules that are currently defined).

kjbracey-arm replied:
IME, AStyle gets it right 99% of the time, at least for normal C code. This sort of complex C++ templatery is probably pushing it too far though.

I'd be fine with having the pad rules as long as errors could be cleaned up manually and not cause a total CI blockage if AStyle sees something it disagrees with in future. (Can you manage to make the CI only check areas touched by diffs, and not recomplain about existing code?)

0xc0170 replied:
@kjbracey-arm That would be best approach (using git to get list of files that are changed in the PR). 
I would like to address some comments here, but the best way to go forward would be to find a best match of astyle that works on our code base C/C++ and use it for all new files (using git to return what files are changed).  We can do cleanup of the codebase even separately

geky commented on events/equeue/equeue.c:
```diff
@@ -245,20 +256,21 @@ static int equeue_enqueue(equeue_t *q, struct equeue_event *e, unsigned tick) {
+            (q->queue == e && !e->sibling)) {
         q->background.update(q->background.timer,
-                equeue_clampdiff(e->target, tick));
+                             equeue_clampdiff(e->target, tick));
```
What rule is in effect here? 3 tabs after newlin in an expression? Why are if statements different?

geky replied:
Then what's going on with this line? https://github.com/ARMmbed/mbed-os/pull/6014#discussion-diff-167327409R303

This seems like it will push lines really far off the page, especially with the longer namespaced names in C++.

geky commented on events/equeue/equeue_posix.c:
```diff
@@ -25,33 +25,39 @@
     struct timeval tv;
     gettimeofday(&tv, 0);
-    return (unsigned)(tv.tv_sec*1000 + tv.tv_usec/1000);
+    return (unsigned)(tv.tv_sec * 1000 + tv.tv_usec / 1000);
```
This code just go harder to understand (where's the precedence?). Is it possible to make astyle be more lax around expression spacing?

0xc0170 replied:
Yes there is a rule. To answer precedence - use braces.
we do spaces around operators

geky replied:
IMO this is more harmful than helpful (what do standard spaces around operators help with?) but fair point on using braces.

geky commented on events/equeue/tests/prof.c:
```diff
@@ -41,92 +41,95 @@ static prof_cycle_t prof_iterations;
+        \
+        printf("\n");                                                           \
+        res;                                                                    \
+    })
```
Need to fix backslashes. I'll see if I can put together a script to do this.

geky commented on features/filesystem/bd/ChainingBlockDevice.h:
```diff
@@ -62,10 +61,9 @@ class ChainingBlockDevice : public BlockDevice
      */
     template <size_t Size>
-    ChainingBlockDevice(BlockDevice *(&bds)[Size])
+    ChainingBlockDevice(BlockDevice * (&bds)[Size])
```
This looks like it broke pointer spacing.

geky commented on features/filesystem/bd/MBRBlockDevice.cpp:
```diff
@@ -67,22 +67,22 @@ static void tochs(uint32_t lba, uint8_t chs[3])
-    struct mbr_table *table = reinterpret_cast<struct mbr_table*>(
-            &buffer[buffer_size - sizeof(struct mbr_table)]);
+    struct mbr_table *table = reinterpret_cast<struct mbr_table *>(
+                                  &buffer[buffer_size - sizeof(struct mbr_table)]);
```
Why was this indention level chosen?

0xc0170 replied:
This is the default behavior for the style:
(`min-conditional-indent=1`) and not defined (`indent-after-parens`)

With `--indent-after-parens` defined:
```
@@ -82,7 +82,7 @@ static int partition_absolute(
     }
 
     struct mbr_table *table = reinterpret_cast<struct mbr_table *>(
-                                  &buffer[buffer_size - sizeof(struct mbr_table)]);
+            &buffer[buffer_size - sizeof(struct mbr_table)]);
     if (table->signature[0] != 0x55 || table->signature[1] != 0xaa) {
         // Setup default values for MBR
         table->signature[0] = 0x55;
@@ -149,7 +149,7 @@ int MBRBlockDevice::partition(BlockDevice *bd, int part, uint8_t type, bd_addr_t
 }
 
 int MBRBlockDevice::partition(BlockDevice *bd, int part, uint8_t type,
-                              bd_addr_t start, bd_addr_t stop)
+    bd_addr_t start, bd_addr_t stop)
 {
     int err = bd->init();
     if (err) {
@@ -204,7 +204,7 @@ int MBRBlockDevice::init()
 
     // Check for valid table
     struct mbr_table *table = reinterpret_cast<struct mbr_table *>(
-                                  &buffer[buffer_size - sizeof(struct mbr_table)]);
+            &buffer[buffer_size - sizeof(struct mbr_table)]);
     if (table->signature[0] != 0x55 || table->signature[1] != 0xaa) {
         delete[] buffer;
         return BD_ERROR_INVALID_MBR;
```

0xc0170 replied:
I pushed an update to see the option actually in place, see https://github.com/ARMmbed/mbed-os/pull/6014/commits/14ea87da959791cfa0395ca6f3faf192ca51d282

geky commented on features/filesystem/fat/FATFileSystem.cpp:
```diff
@@ -142,11 +139,11 @@ DWORD get_fattime(void)
+           | (DWORD)(ptm->tm_mday) << 16
+           | (DWORD)(ptm->tm_hour) << 11
+           | (DWORD)(ptm->tm_min) << 5
+           | (DWORD)(ptm->tm_sec / 2);
```
Hm, this just got harder to read. Is it possible to make astyle more lax in this situation?

geky commented on features/filesystem/littlefs/LittleFileSystem.cpp:
```diff
@@ -26,78 +26,100 @@ extern "C" {
+        case LFS_ERR_NOMEM:
+            return -ENOMEM;
+        default:
+            return err;
```
Do we have a stance on single line switch statements? These are pretty common in embedded programming and more difficult to read/edit in this format.

0xc0170 replied:
One line one statement.

kjbracey-arm replied:
Formally, that's a *labelled-statement* occupying two lines. Squish them together and you'd have one line one *labelled-statement*.

FWIW, I'm might be inclined to squish in those circumstances, but it's a very special case for exactly 1 label per 1 statement. Forcing the line break feels consistent with the logic for "single-line conditional statements are braced as if multiline".

geky replied:
Ok, sounds like I'm in the minority so let's keep it as is

geky commented on features/filesystem/littlefs/LittleFileSystem.cpp:
```diff
@@ -275,7 +300,7 @@ int LittleFileSystem::reformat(BlockDevice *bd)
 
     int err = LittleFileSystem::format(bd,
-            _read_size, _prog_size, _block_size, _lookahead);
+                                       _read_size, _prog_size, _block_size, _lookahead);
```
What in the world's going on with this line?

bulislaw commented:
That's going to break all of the PRs, feature branches and code that's being baked. Not to mention big-big issues with all the rebases.

kjbracey-arm commented on events/Event.h:
```diff
@@ -287,159 +287,159 @@ class Event<void()> {
      */
     template <typename T, typename R, typename B0>
-    Event(EventQueue *q, T *obj, R (T::*method)(B0), B0 b0) {
+    Event(EventQueue *q, T *obj, R(T::*method)(B0), B0 b0) {
```
Is it just misunderstanding the syntax and thinking it's a function call?

0xc0170 replied:
Thanks pan, will check these type of changes

kjbracey-arm commented:
@bulislaw   There is an easy-ish way to deal with PRs/feature branches - used it last time I did this sort of thing. You can reformat and rebase each commit of the PR/feature branch with the same astyle settings. The core magic command is `git filter-branch --tree-filter astyle (assorted parameters)` 

I've got step-by-step instructions on an internal Trac, which we could publish.  Mind you, that page also started with:

> "Reformatting is bad. Don't. Never reformat code unnecessarily. Reformatting under any source control causes all sorts of headaches - it's hard to locate real differences, and conflicts arise when more than one developer is working on a file.

Having said that, I did do a reformat (to the Nanostack repo), but only after I had established and documented those "reformat your branch" instructions. Also, we needed to make a big shift over to quite-different mbed OS style, and there wasn't anywhere near as much ongoing work.

I agree there should be flexibility on enforcement. AStyle sometimes gets it straight wrong due to syntax misunderstanding, and sometimes its broader formatting choices on a multi-line scale are not ideal. Had to do a few manual cleanups after autoformatting last time.

geky commented:
Would we be interested in a bot that just comments on new rule violations?

geky commented on events/EventQueue.cpp:
```diff
@@ -19,35 +19,42 @@
-EventQueue::EventQueue(unsigned event_size, unsigned char *event_pointer) {
+EventQueue::EventQueue(unsigned event_size, unsigned char *event_pointer)
+{
     if (!event_pointer) {
```
My complaint is that squiggly brackets are still inconsistent.

We need to choose one of these:
``` cpp
class Thing {
    int dothing() {
        if (thing) {
            blahblahblah;
        }
    }
};
```

``` cpp
class Thing
{
    int dothing()
    {
        if (thing)
        {
            blahblahblah;
        }
    }
};
```

Not both, what's the point of this?

``` cpp
class Thing {
    int dothing()
    {
        if (thing) {
            blahblahblah;
        }
    }
};
```

0xc0170 replied:
This is how they have been, attached but not for functions, see our coding style since mbed 2.

geky replied:
My point is that even though it's in our style guide, we're not using it consistently (see above numbers). Perhaps because it's confusing.

0xc0170 replied:
You are the first to mention this that I am aware of. 

Time to set it correctly now in all files and have it checked.

0xc0170 commented:
I reviewed options to answer some questions about styling changes here.
 
What is outstanding is how to align multiple lines - we did not cover it in the coding style guide, however defined astyle command had it by default set up - same as we have it here now - no changes there (as noted some *& might have problems with spacing that is AStyle issue - few instances in this PR should not block this effort, can be dealt separately when they arise). By default, with what we had defined for astyle, it would align multiple line statements and indented lines after its parents.

Changes like this:
```
     int err = LittleFileSystem::format(bd,
-            _read_size, _prog_size, _block_size, _lookahead);
+                                       _read_size, _prog_size, _block_size, _lookahead);


         q->background.update(q->background.timer,
-                equeue_clampdiff(e->target, tick));
+                             equeue_clampdiff(e->target, tick));
```


I defined a new option to see the changes to address some comments above, please review 14ea87d. these 2 changes would become:

```
 int LittleFileSystem::format(BlockDevice *bd,
-                             lfs_size_t read_size, lfs_size_t prog_size,		 
+    lfs_size_t read_size, lfs_size_t prog_size,
-                             lfs_size_t block_size, lfs_size_t lookahead)		 
+    lfs_size_t block_size, lfs_size_t lookahead)

     if ((q->background.update && q->background.active) &&
-            (q->queue == e && !e->sibling)) {		 
+        (q->queue == e && !e->sibling)) {
          q->background.update(q->background.timer,		          
-                             equeue_clampdiff(e->target, tick));		 
+            equeue_clampdiff(e->target, tick));
```

0xc0170 commented:
/morph build

0xc0170 commented:
The failure is fixed (I missed one common folder for renesas). restarting the build

/morph build

0xc0170 commented:
/morph build

0xc0170 commented:
/mbed test

geky commented:
> That's going to break all of the PRs, feature branches and code that's being baked.

IMO this should be the **last** pr merged for 5.8. Preferably after the code freeze and as the first action of making the release. This worked well for the last change of this scale (https://github.com/ARMmbed/mbed-os/pull/2878).

Otherwise the rebase conflicts will cause a lot of problems for the prs trying to get in on 5.8. It will be a pain to rebase this pr, but it will be easier to rebase 1 pr rather than n prs.

cmonr commented:
FYI @0xc0170 @adbridge ^^^

0xc0170 commented:
@geky 👍 
Yes, but not certain if this lands for 5.8, I'll rebase today with some rework.  Will provide update

0xc0170 commented:
Updated. The style was simplified a bit (removed ~3 setting), it should now match the rules in our coding style (for reference our older one https://os.mbed.com/teams/SDK-Development/wiki/mbed-sdk-coding-style)

I run this new update on the entire codebase besides those folder in the ignore list. New addition there, targets were added for now, they are skipped from the style check.

0xc0170 commented:
I fixed the options, now they are compatible with astyle 3.1 and older version (tested down to 2.05.01). 2.05.01 (almost 4 years old) produces very similar result, just does not have a fix for new/delete spacing issue from what I noticed, thus the diff is a bit different.  At least the options we define here are compatible.

Rebased to the latest master, fixed the style issues for the new files

0xc0170 commented:
/morph build

0xc0170 commented:
Not certain about the hook `travis-ci/astyle — Local astyle testing is in progress` . I restarted it 3x already, always green, and the only hook changed here is travis.

This should be up for review

geky commented:
It looks like an exit in the astyle script caused the Travis process to exit completely instead of moving to the next stage and updating the status : /

I went ahead and pushed a fix (subshell) feel free to squash or change it.

geky commented:
![image](https://user-images.githubusercontent.com/1075160/37428999-01670c06-279c-11e8-9b20-d28e11418e9a.png)
@0xc0170, looks like it already caught something :stuck_out_tongue:  

Also damn you grammar!

0xc0170 commented:
This is good (showcase how it works when it fails)! I'll rebase to fix the code style issue.

cmonr commented:
@0xc0170 Since this touches a whole bunch of files, I'm thinking this will need to either
1) come in around 5.9.0-rc1 -and/or-
2) be synchronized with other PR activity from the next time you rebase to when this PR comes in in relation to other PRs.

It looks like it will be ready after another rebase.

0xc0170 commented:
> come in around 5.9.0-rc1 -and/or-

I fear that this might be the case. However if the style is approved, still this can land partially (astyle check for new code only - so we start slowly adapting the new code - should be fairly simple to update and send via a new PR), and we update the older code for 5.9. Open for suggestions.

cmonr commented:
@0xc0170 I think the style checks as they are now are fine, and think everyone else is now onboard with them (at least, no one as raised any other concerns with them).

I think we could take the following approach:
1. Start by adding the astyle check into Travis, but don't have it actually break builds. It should only report warnings, but always report a "Success" status.
2. Using the list above, cherry pick this PR's commits until the need to rebase goes away. This should help get us get a bulk of the way there without needing to rebase and update the commits. The caveat with this is that PRs that are currently open will need to be rebased once this step is completed (most likely).
3. Add an astyle commit one by one, until the rest of the codebase is up to date.

(In case it wasn't clear, each of those above steps would be an individual PR. #3 would involve multiple PRs.)

geky commented:
+1 for the astyle travis check as a warning until 5.9!

screamerbg commented:
@0xc0170 +1 for the idea and the energy put in this.

To make this less disruptive I think that we need to:
1. enable astyle check as info (not warnings nor errors)
2. get OS code piece by piece aligned/update with astyles (not as one mindblowing PR)
3. finalize OS code transition to astyle to a point where there are no astyle issues
4. once complete, get alignment with Mbed Partner teams and expand astyle to their HAL ports
5. switch from info to warnings
6. beyond mbed OS 5.9 switch to errors

0xc0170 commented:
This serves as an example for astyle to see all the code that is affected. I am preparing smaller one that is only adding travis and astyle config. It will check only files that has changed, once I have it running will reference here and close this one.

0xc0170 commented:
> enable astyle check as info (not warnings nor errors)

This should do it #6590 .  Once this one is in, I'll start with the code base one by one

I am closing this for now
